src/managers/ThreadManager.js

```diff
@@ -0,0 +1,218 @@
+'use strict';
+
+const BaseManager = require('./BaseManager');
+const { TypeError } = require('../errors');
+const ThreadChannel = require('../structures/ThreadChannel');
+const { Collection } = require('../util/Collection');
+
+/**
+ * Manages API methods for ThreadChannels and stores their cache.
+ * @extends {BaseManager}
+ */
+class ThreadManager extends BaseManager {
+  constructor(channel, iterable) {
+    super(channel.client, iterable, ThreadChannel);
+
+    /**
+     * The channel this Manager belongs to
+     * @type {NewsChannel|TextChannel}
+     */
+    this.channel = channel;
+  }
+
+  /**
+   * The cache of this Manager
+   * @type {Collection<Snowflake, ThreadChannel>}
+   * @name ThreadManager#cache
+   */
+
+  add(thread) {
+    const existing = this.cache.get(thread.id);
+    if (existing) return existing;
+    this.cache.set(thread.id, thread);
+    return thread;
+  }
+
+  /**
+   * Data that can be resolved to give a Thread Channel object. This can be:
+   * * A ThreadChannel object
+   * * A Snowflake
+   * @typedef {ThreadChannel|Snowflake} ThreadChannelResolvable
+   */
+
+  /**
+   * Resolves a ThreadChannelResolvable to a Thread Channel object.
+   * @method resolve
+   * @memberof ThreadManager
+   * @instance
+   * @param {ThreadChannelResolvable} thread The ThreadChannel resolvable to resolve
+   * @returns {?ThreadChannel}
+   */
+
+  /**
+   * Resolves a ThreadChannelResolvable to a thread channel ID string.
+   * @method resolveID
+   * @memberof ThreadManager
+   * @instance
+   * @param {ThreadChannelResolvable} thread The ThreadChannel resolvable to resolve
+   * @returns {?Snowflake}
+   */
+
+  /**
+   * Creates a new thread in the channel.
+   * @param {string} name The name of the new Thread
+   * @param {number} autoArchiveDuration How long in minutes before the thread is automatically archived,
+   * one of `60`, `1440`, `4320`, or `10080`
+   * @param {Object} options Options
+   * @param {MessageResolvable} [options.startMessage] The message to start a public or news thread from,
+   * creates a private thread if not provided
+   * @param {string} [options.reason] Reason for creating the thread
```

Comment:
Yeah I know but wouldn’t it be better to make it an actual typedef? Not sure since the two options are used somewhat inconsistently throughout the library